### ISSUE TITLE
JSON ERROR >= 4 Syntax Error "wrong formatted JSON String"

### ISSUE THREAD

**Issue by smith844** [reloxx13/TasmoAdmin#228]
I am running 10 devices already (mixture of SONOFF and ESP-01 with TASMOTA) which i added once tasmo admin was added to hassio (running on Raspi 3+).  when i try to add further devices I get an error "Device Not Found" then : 
JSON ERROR => 4: Syntax error
Please copy the whole error message and copy it in a new issue on GitHub. Somehow the Tasmota Firmware answered a wrong formatted JSON string which needs to get adjusted in TasmoAdmin.
JSON Answer => {"Version":"6.1.1","BuildDateTime":"2018-09-14T19:17:05","Boot":31,"Core":"2_4_0_RC2","SDK":"2.1.0(7106d38)"},"StatusLOG":{"SerialLog":4,"WebLog":4,"SysLog":4,"LogHost":"192.168.1.","LogPort":514,"SSId":["BTHub6-####","BTHub6-####"],"TelePeriod":10,"SetOption":["00000009","55818000"]},"StatusMEM":{"ProgramSize":554,"Free":448,"Heap":6,"ProgramFlashSize":1024,"FlashSize":1024,"FlashMode":3,"Features":["00000809","0FDAE794","00000000","23B6179E","00000000"]},"StatusNET":{"Hostname":"homeassistant/freezer-0850","IPAddress":"192.168.1.104","Gateway":"192.168.1.254","Subnetmask":"255.255.255.0","DNSServer":"192.168.1.254","Mac":"##:##:##:##:##:##","Webserver":2,"WifiConfig":3},"StatusMQT":{"MqttHost":"192.168.1.205","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_0F8352","MqttUser":"homeassistant","MqttType":1,"MAX_PACKET_SIZE":1000,"KEEPALIVE":15},"StatusTIM":{"UTC":"Sat Sep 15 09:29:32 2018","Local":"Sat Sep 15 10:29:32 2018","StartDST":"Sun Mar 25 02:00:00 2018","EndDST":"Sun Oct 28 03:00:00 2018","Timezone":1,"Sunrise"

that is grabbed straight from screen.
so my issues are:
Can someone shed some light on the error.
Is there an error log somewhere I can cut and paste complete ?

**Comment by smith844** [comment]
JSON ERROR => 4: Syntax error
Please copy the whole error message and copy it in a new issue on GitHub. Somehow the Tasmota Firmware answered a wrong formatted JSON string which needs to get adjusted in TasmoAdmin.
JSON Answer => {"SerialLog":4,"WebLog":4,"SysLog":4,"LogHost":"192.168.1.205","LogPort":514,"SSId":["BTHub6-XSQX","BTHub6-XSQX"],"TelePeriod":10,"SetOption":["00000009","55818000"]},"StatusMEM":{"ProgramSize":554,"Free":448,"Heap":6,"ProgramFlashSize":1024,"FlashSize":1024,"FlashMode":3,"Features":["00000809","0FDAE794","00000000","23B6179E","00000000"]},"StatusNET":{"Hostname":"homeassistant/freezer-0850","IPAddress":"192.168.1.104","Gateway":"192.168.1.254","Subnetmask":"255.255.255.0","DNSServer":"192.168.1.254","Mac":"60:01:94:0F:83:52","Webserver":2,"WifiConfig":3},"StatusMQT":{"MqttHost":"192.168.1.205","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_0F8352","MqttUser":"homeassistant","MqttType":1,"MAX_PACKET_SIZE":1000,"KEEPALIVE":15},"StatusTIM":{"UTC":"Sat Sep 15 09:51:05 2018","Local":"Sat Sep 15 10:51:05 2018","StartDST":"Sun Mar 25 02:00:00 2018","EndDST":"Sun Oct 28 03:00:00 2018","Timezone":1,"Sunrise":"06:45","Sunset":"19:26"},"StatusSNS":{"Time":"2018-09-15T10:51:05","DS18B20":{"Temperature":-7.6},"TempUnit":"C"},"StatusSTS":{"Time":"2018-09-15T10:51:05","Uptime":"0T00:00:13","Vcc":3.388,"Wifi":{"AP":1,"SSId":"BTHub6-XSQX","RSSI":50,"APMac":"AC:84:C9:AD:90:EA"}}}

That is, as far as I can see, the whole of the error, some of the original post was missing

**Comment by smith844** [comment]
Thanks very much, both for your swift reply and the obvious work that you have put into a great program.
I checked a number of devices that I had flashed and none of them had a complete JSON string.  I updated them with 6.2.1 and this was rectified.  I could then add them in tasmoadmin as normal.  I use Arduino IDE to upload the tasmota software and the uploads seemed to work but when checked with status 200 the string was always missing characters.  I updated the source folder on Arduino IDE installation and flashed another device and this was fine.  So it appears the issue was more to do with Arduino and Sonoff than Tasmoadmin. 
Thank you again.
